src/zip_node.js

```diff
@@ -43,19 +43,27 @@ const addStat = async function (srcFile) {
   return { srcFile, stat }
 }
 
-const zipJsFile = function ({ srcFile, commonPrefix, archive, stat }) {
-  const filename = normalizeFilePath(srcFile, commonPrefix)
+const zipJsFile = function ({ srcFile, commonPrefix, additionalPrefixes, archive, stat }) {
+  const filename = normalizeFilePath(srcFile, commonPrefix, additionalPrefixes)
   addZipFile(archive, srcFile, filename, stat)
 }
 
 // `adm-zip` and `require()` expect Unix paths.
 // We remove the common path prefix.
 // With files on different Windows drives, we remove the drive letter.
-const normalizeFilePath = function (path, commonPrefix) {
+const normalizeFilePath = function (path, commonPrefix, additionalPrefixes = []) {
   const pathA = normalize(path)
-  const pathB = pathA.replace(commonPrefix, `${ZIP_ROOT_DIR}${sep}`)
-  const pathC = unixify(pathB)
-  return pathC
+
+  // additional prefixes are used to write the dependency in the correct location in the zip file
+  // e.g. if we resolved a dependency from .netlify/plugins/node_modules/<path>, the target path should be
+  // src/node_modules/<path>
+  const pathB = additionalPrefixes.reduce(
+    (acc, prefix) => acc.replace(normalize(prefix), `${ZIP_ROOT_DIR}${sep}node_modules`),
```

Comment:
Would `acc.replace(normalize(prefix), 'node_modules'),` be enough?
Since `${ZIP_ROOT_DIR}${sep}` is added by the next statement.

Reply:
There's actually a bug here I noticed thanks to your comment.
The prefixes don't have to be under the `srcDir` (the now removed use case from #116), so the `commonPrefix` approach won't work in that case.

We could either think of a way to support it or define that `nodeResolvePaths` are relative to the `srcDir`. The latter would make it easier to provide a default. WDYT?

Reply:
`commonPrefix` should actually work even when one of the required file paths is not under the repository root. If I'm not mistaken, the logic would work even when requiring a different top-level directory (in that case, `commonPrefix` would be `/`).

Would adding `prefixes` to the `commonPathPrefix()` call work?